### PR TITLE
GH-439: Add 'transacted' to Rabbit Producer Props

### LINKED ISSUE #401: Remove code in binders related to original use in Spring XD
In https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java#L469  there is code that creates a new classloader that was originally done to avoid errors such as https://jira.spring.io/browse/XD-2509

As there are not multiple classloaders in spring-cloud-stream's runtime, it is quite likely this code should be removed.

### LINKED ISSUE #439: Rabbit Binder Transacted Producer

### PR DESCRIPTION
**Pull request by garyrussell**
Fixes #439
Resolves #401 

I suggest reviewing on GitHub with `?w=1` in the URL to ignore whitespace changes - lots of them a big indented try/finally is removed.

### CODE REVIEW

File: spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java [hunk 1]
```diff
@@ -291,70 +291,63 @@ public void onInit() {
...
-			if (maxAttempts > 1 || properties.isRepublishToDlq()) {
-				RetryOperationsInterceptor retryInterceptor = RetryInterceptorBuilder.stateless()
-						.maxAttempts(maxAttempts)
-						.backOffOptions(properties.getBackOffInitialInterval(),
-								properties.getBackOffMultiplier(),
-								properties.getBackOffMaxInterval())
-						.recoverer(determineRecoverer(name, properties.getPrefix(), properties.isRepublishToDlq()))
-						.build();
-				listenerContainer.setAdviceChain(new Advice[] { retryInterceptor });
+		ClassUtils.overrideThreadContextClassLoader(SimpleMessageListenerContainer.class.getClassLoader());
```
Comment (mbogoevici) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java]: We can do away with overriding the TCCL here, too, I think, especially as it is not restored at the end of the method. This has little relevance for Boot applications anyway.

**Comment by mbogoevici** [comment]
Everything else LGTM. Will merge.

**Comment by mbogoevici** [comment]
Merged w/https://github.com/spring-cloud/spring-cloud-stream/commit/fa035296f2338d024fcf5f1178d26e69d3a4c360

File: spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java (hunk 1 above)
Comment (garyrussell) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java]: Thanks; good catch!

**Pull request closed without merging**